pkg/cluster/fixssh.go

```diff
@@ -33,21 +31,41 @@ func (m *manager) fixSSH(ctx context.Context) error {
 		return fmt.Errorf("unknown architecture version %d", m.doc.OpenShiftCluster.Properties.ArchitectureVersion)
 	}
 
-	lb, err := m.loadBalancers.Get(ctx, resourceGroup, lbName, "")
+	resourceGroup := stringutils.LastTokenByte(m.doc.OpenShiftCluster.Properties.ClusterProfile.ResourceGroupID, '/')
+
+	lb, err := m.checkAndUpdateLB(ctx, resourceGroup, lbName)
 	if err != nil {
+		m.log.Warnf("Failed checking and Updating Load Balancer with error: %s", err)
 		return err
 	}
 
-	changed := updateLB(&lb)
+	err = m.checkandUpdateNIC(ctx, resourceGroup, infraID, lb)
+	if err != nil {
+		m.log.Warnf("Failed checking and Updating Network Interface with error: %s", err)
+		return err
+	}
+	return nil
+}
+
+func (m *manager) checkAndUpdateLB(ctx context.Context, resourceGroup string, lbName string) (lb mgmtnetwork.LoadBalancer, err error) {
+	lb, err = m.loadBalancers.Get(ctx, resourceGroup, lbName, "")
+	if err != nil {
+		return lb, err
+	}
+
+	lbChanged := m.updateLB(ctx, &lb)
 
-	if changed {
+	if lbChanged {
```

Comment:
It would read better if we dropped the `lbChanged` variable altogether:
```go
if m.updateLB(ctx, &lb) {
```

Reply:
Dropped `lbchanged` variable altogether as you suggested.